File: ObserverV2C#/Observer/src/DigitalClock.cs
using Observer.include;
using System;
using System.Collections.Generic;
using System.Text;

namespace Observer.src
{
    public class DigitalClock: IDigitalClock {

        public DigitalClock(IClockTimer clockTimer) {
            timer = clockTimer;
            timer.GetSubscriber().Attach(this);
        }

        public void Update()
        {
            Draw();
        }

        public void Draw()
        {
            string hours = timer.GetHour().ToString();
            string minutes = timer.GetMinute().ToString();
            string seconds = timer.GetSecond().ToString();

            if (hours.Length == 1) { hours = "0" + hours; }
            if (minutes.Length == 1) { minutes = "0" + minutes; }
            if (seconds.Length == 1) { seconds = "0" + seconds; }
        }


        private IClockTimer timer;

    }
}

File: ObserverV2Java/ObserverJava/src/src/DigitalClock.java
package src;

import include.IClockTimer;
import include.IDigitalClock;

public class DigitalClock implements IDigitalClock {

    public DigitalClock(IClockTimer clockTimer) {
        timer = clockTimer;
        timer.getSubscriber().attach(this);
    }

    public void update() {
        draw();
    }

    public void draw() {
        String hours = String.valueOf(timer.getHour());
        String minutes = String.valueOf(timer.getMinute());
        String seconds = String.valueOf(timer.getSecond());

        if (hours.length() == 1) {
            hours = "0" + hours;
        }
        if (minutes.length() == 1) {
            minutes = "0" + minutes;
        }
        if (seconds.length() == 1) {
            seconds = "0" + seconds;
        }
        System.out.println("I am Digital: " + hours + ":" + minutes + ":" + seconds);
    }

    private IClockTimer timer;

}
